qiskit_machine_learning/algorithms/distribution_learners/__init__.py

```diff
@@ -20,6 +20,17 @@
     QuantumGenerator,
     QGAN,
 )
+from ...deprecation import warn_deprecated, DeprecatedType, MachineLearningDeprecationWarning
+
+warn_deprecated(
```

Comment:
I am not sure with this. In algorithms.init file we import these so I think users are going to see messages whether they use the classes or not. We may need to put deprecation more local in constructors. @manoelmarques any thoughts here. If we do that we may also want to change the first line of the class doctring to prefix it with Deprecated:  so its evident in the docs. Since all the classes are documented in algorithms init though an alternative/addition would be to state they are deprecated there too.

Also we normally adjust unit tests of deprecated code, since we know it will emit such messages, so as to restrict what comes out in CI to deprecations we should be taking care of.

Reply:
If `distribution_learners` folder is included in any `init` above it, then unfortunately the deprecation can't be at the `init` level otherwise any unrelated import will import it and show the deprecation msg. In  that case it needs to be in the `init` of each class inside the folder.

Reply:
Moved deprecations to the class level.